app/models/piece.rb

```diff
@@ -64,7 +64,11 @@ def is_obstructed?(row_final, col_final)
 
       #diagonal case
     elsif ((row_final - row_init).to_f/(col_final - col_init).to_f ).abs == 1
-      return Piece.exists?(position_row: row_range, position_column: col_range, game_id: game_id)
+      i = 0
```

Comment:
Good job fixing this bug. I tested with rspec the pieces located off the diagonal, and got the results that you mentioned.